### PR TITLE
Draft PR rework

### PR DESCRIPTION
**Pull request by wilmaier**

### CODE REVIEW

File: packages/itwin/map-layers/src/ui/FeatureInfoUiItemsProvider.tsx [hunk 1]
```diff
@@ -17,15 +17,40 @@ import {
...
 
 import { MapLayersUI } from "../mapLayers";
 import { MapFeatureInfoWidget } from "./widget/FeatureInfoWidget";
 import type { MapFeatureInfoOptions } from "./Interfaces";
 import { MapLayersSyncUiEventId } from "../MapLayersActionIds";
-import type { MapImagerySettings } from "@itwin/core-common";
+
+const supportsMapFeatureInfo = (vp: ScreenViewport, mapLayerProps: MapLayerProps[]): boolean => {
+  for (let mapLayerIndex = 0; mapLayerIndex < mapLayerProps.length; mapLayerIndex++) {
+    const bgLayerProvider = vp.getMapLayerImageryProvider({ index: mapLayerIndex, isOverlay: false });
```
Comment (mdastous-bentley) [inline packages/itwin/map-layers/src/ui/FeatureInfoUiItemsProvider.tsx:32]: `isOverlay` should be a parameter here, and not hardcoded to false?

File: packages/itwin/map-layers/src/ui/FeatureInfoUiItemsProvider.tsx [hunk 2]
```diff
@@ -39,68 +64,16 @@ export const getMapFeatureInfoToolItemDef = (): ToolItemDef =>
...
-
-  public constructor(private _featureInfoOpts: MapFeatureInfoOptions) {
-    IModelApp.viewManager.onViewOpen.addOnce((vp: ScreenViewport) => {
-      const handleMapImageryChanged = (_args: Readonly<MapImagerySettings>) => {
-        FeatureInfoUiItemsProvider.loadMapLayerProviders();
-      };
-      const handleDisplayStyleChanged = () => {
-        FeatureInfoUiItemsProvider.loadMapLayerProviders();
-      };
-      vp.displayStyle.settings.onMapImageryChanged.addListener(handleMapImageryChanged);
```
[the remark this answers is not in the document]
Comment (wilmaier) [reply packages/itwin/map-layers/src/ui/FeatureInfoUiItemsProvider.tsx:70]: Yes the event handlers only were needed to keep the static array of providers up to date.  Now they are just retrieved as needed.

File: packages/itwin/map-layers/src/ui/FeatureInfoUiItemsProvider.tsx (hunk 1 above)
Comment (wilmaier) [reply packages/itwin/map-layers/src/ui/FeatureInfoUiItemsProvider.tsx:32]: Yes I overlooked that during the rework.

**Review by mdastous-bentley — APPROVED** [review]

**Comment by aruniverse** [comment]
What is this pr?
This title and description are completely useless and doesn't help anyone

**Comment by wilmaier** [comment]
Good point.  I just used this draft PR to do some rework of the original PR before merging it.

**Comment by mdastous-bentley** [comment]
@wilmaier Does this need to be backported on any release branch ?

**Comment by aruniverse** [comment]
> @wilmaier Does this need to be backported on any release branch ?

we will not be doing any backports in this repo. they need to update to the latest

**Pull request merged**